scripts/scanDependents/scans.ts

```diff
@@ -84,3 +106,99 @@ export async function scanDependentDependencyVersions(
     }),
   }));
 }
+
+export interface SearchRepositoriesOptions extends RequestOptions, LoggingOptions {
+  repositories: string[];
+  searchString: string;
+  fileExtensions?: string[];
+}
+
+export async function searchRepositories({
+  repositories,
+  searchString,
+  fileExtensions,
+  verbose,
+  ...requestOpts
+}: SearchRepositoriesOptions) {
+  // There are two types of requests that we need to set up for GitHub
+  const apiRequest = getApiRequest(requestOpts);
+
+  // Make request to Github search API
+  const repoQualifiers = repositories.map((repo) => `repo:${repo}`).join(' ');
+  const extensionQualifiers = fileExtensions?.map((ext) => `extension:${ext}`).join(' ') ?? '';
+  const q = `${searchString} ${repoQualifiers} ${extensionQualifiers}`;
+
+  if (verbose) {
+    console.log('Performing the following code search:');
+    console.log('');
+    console.log(chalk.cyan(q));
+    console.log('');
+    process.stdout.write('Fetching and tabulating results...');
+  }
+
+  const repositoryResults: { [repoName: string]: number } = {};
+  let page = 1;
+  const perPage = 50;
+  let itemsProcessed = 0;
+  let searchResponse;
+  do {
+    searchResponse = await apiRequest('GET /search/code', { q, page, per_page: perPage });
+    for (const item of searchResponse.data.items) {
+      const repo = item.repository.full_name;
```

Comment:
I don't know what this would look like but would be nice to get the URL's back from the request so we could click around to the references and maybe suggest changes in those cases.. Not in-scope for this one but maybe a future addition!